src/ReactART.js

```diff
@@ -329,315 +88,446 @@ const NodeMixin = {
       );
 
-      this._oldPath = path;
-      this._oldDelta = path.delta;
+      this._prevPath = path;
+      this.prevDelta = path.delta;
     }
-
-    this.applyRenderableProps(oldProps, props);
   }
+}
 
-});
-
-// Text
+class Text extends RenderableNode {
+  constructor(props) {
+    const string = childrenAsString(props.children);
+    const instance = Mode.Text(
+      string,
+      props.font,
+      props.alignment,
+      props.path,
+    );
 
-const Text = createComponent('Text', RenderableMixin, {
+    super(TYPES.TEXT, instance, props);
 
-  construct: function(element) {
-    this._currentElement = element;
-    this._oldString = null;
-  },
+    this._currentString = string;
+  }
 
-  mountComponent: function(
-    transaction,
-    nativeParent,
-    nativeContainerInfo,
-    context
+  applyProps(
+    props,
+    prevProps = {},
   ) {
-    const props = this._currentElement.props;
-    const newString = childrenAsString(props.children);
-    this.node = Mode.Text(newString, props.font, props.alignment, props.path);
-    this._oldString = newString;
-    this.applyRenderableProps(emptyObject, props);
-    return this.node;
-  },
+    super.applyProps(props, prevProps);
+
+    const string = childrenAsString(props.children);
+
+    if (
+      this._currentString != string ||
+      !this._isSameFont(props.font, prevProps.font) ||
+      props.alignment !== prevProps.alignment ||
+      props.path !== prevProps.path
+    ) {
+      this.instance.draw(
+        string,
+        props.font,
+        props.alignment,
+        props.path,
+      );
 
-  isSameFont: function(oldFont, newFont) {
+      this._currentString = string;
+    }
+  }
+
+  _isSameFont(oldFont, newFont) {
     if (oldFont === newFont) {
       return true;
-    }
-    if (typeof newFont === 'string' || typeof oldFont === 'string') {
+    } else if (
+      typeof newFont === 'string' ||
+      typeof oldFont === 'string'
+    ) {
       return false;
+    } else {
+      return (
+        newFont.fontSize === oldFont.fontSize &&
+        newFont.fontStyle === oldFont.fontStyle &&
+        newFont.fontVariant === oldFont.fontVariant &&
+        newFont.fontWeight === oldFont.fontWeight &&
+        newFont.fontFamily === oldFont.fontFamily
+      );
     }
-    return (
-      newFont.fontSize === oldFont.fontSize &&
-      newFont.fontStyle === oldFont.fontStyle &&
-      newFont.fontVariant === oldFont.fontVariant &&
-      newFont.fontWeight === oldFont.fontWeight &&
-      newFont.fontFamily === oldFont.fontFamily
+  }
+}
+
+class Surface extends Component {
+  componentDidMount() {
+    const { height, width } = this.props;
+
+    this._surface = Mode.Surface(+width, +height, this._tagRef);
+
+    // HACK Simplify appendChild-type methods by mirroring Art instance wrappers
+    const container = {
+      instance: this._surface,
+    };
+
+    this._mountNode = ARTRenderer.mountContainer(
+      this.props.children,
+      container,
```

Comment:
This may be a stupid question...but...why do ReactART components need `context`? The set of components being managed by the `ARTRenderer` only includes the ones explicitly defined in this module, none of which use `context`.